src/ast/module.cc

```diff
@@ -187,11 +188,15 @@ namespace
     }
   }
 
-  bool run_passes(ModulePtr& m, const Passes& passes, err::Errors& err)
+  bool run_passes(
+    ModulePtr& m,
+    const std::string& stopAt,
+    const pass::Passes& passes,
+    err::Errors& err)
   {
     return dfs::post(
       m,
-      [](auto& m, auto& passes, auto& err) {
+      [&stopAt](auto& m, auto& passes, auto& err) {
```

Comment:
I should really rewrite dfs::post to expect the lambda to capture more. It's better to do what you've done here (capture &stopAt) than to thread things through.